baml_language/crates/baml_hir/src/lib.rs

```
@@ -687,16 +741,35 @@ pub fn lower_type_ref(node: &baml_syntax::ast::TypeExpr) -> TypeRef {
         "video" => TypeRef::Video,
         "pdf" => TypeRef::Pdf,
         _ => {
+            // Check for string literal types
+            if (text.starts_with('"') && text.ends_with('"'))
+                || (text.starts_with('\'') && text.ends_with('\''))
+            {
+                // String literal type; extract the value
+                let inner = &text[1..text.len() - 1];
+                return TypeRef::StringLiteral(inner.to_string());
+            }
+
             // Check if it ends with '?' (optional)
             if let Some(inner_text) = text.strip_suffix('?') {
-                let inner = TypeRef::named(inner_text.into());
+                let inner = lower_type_text(inner_text);
                 TypeRef::optional(inner)
             }
             // Check if it ends with '[]' (list)
             else if let Some(inner_text) = text.strip_suffix("[]") {
-                let inner = TypeRef::named(inner_text.into());
+                let inner = lower_type_text(inner_text);
                 TypeRef::list(inner)
             }
+            // Check for boolean literal types
+            else if text == "true" {
+                TypeRef::BoolLiteral(true)
+            } else if text == "false" {
+                TypeRef::BoolLiteral(false)
+            }
+            // Check for integer literal types (for exhaustiveness)
+            else if let Ok(int_val) = text.parse::<i64>() {
+                TypeRef::IntLiteral(int_val)
+            }
             // Otherwise treat as named type
             else {
                 TypeRef::named(text.into())
```

I think float literals are not handled here and they are treated as `TypeRef::Named`?

---

The TODO comment is still relevant here; we should still introduce some sort of helpful diagnostic in the future (or do some magic stuff under the hood). but, in the meantime, ive added an explicit `return TypeRef:Error` if we cannt parse as an `i64` or starts with text (float literal). at least we are not silently failing.

---

fyi: i'll not resolve these comments.